core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java

```diff
@@ -154,11 +146,30 @@ protected void configure(BuildImageCmd buildImageCmd) {
         this.dockerfile.ifPresent(p -> {
             buildImageCmd.withDockerfile(p.toFile());
             dependencyImageNames = new ParsedDockerfile(p).getDependencyImageNames();
+
+            if (dependencyImageNames.size() > 0) {
+                // if we'l be pre-pulling images, disable the built-in pull as it is not necessary and will fail for
+                // authenticated registries
+                buildImageCmd.withPull(false);
+            }
         });
 
         this.buildArgs.forEach(buildImageCmd::withBuildArg);
     }
 
+    private void prePullDependencyImages(Set<String> imagesToPull) {
+        final DockerClient dockerClient = DockerClientFactory.instance().client();
+
+        imagesToPull.forEach(imageName -> {
```

Comment:
nit: why not the good old for-in? :D

Reply:
That's a really good point - I prefer that style anyway, don't know why I did this :D

Reply:
Maybe it was a bit more complex before and involved some `Stream`s or something? :)

Reply:
I blame the month-ago version of me - he had no idea what he was doing 😂